modules/core/include/opencv2/core/utils/logger.hpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_LOGGING_HPP
+#define OPENCV_LOGGING_HPP
+
+#include <iostream>
+#include <sstream>
+#include <limits.h> // INT_MAX
+
+// TODO This file contains just interface part with implementation stubs.
+
+//! @addtogroup core_logging
+// This section describes OpenCV logging utilities.
+//
+//! @{
+
+namespace utils {
+namespace logging {
+
+#define CV_LOG_LEVEL_SILENT 0
+#define CV_LOG_LEVEL_FATAL 1
+#define CV_LOG_LEVEL_ERROR 2
+#define CV_LOG_LEVEL_WARN 3
+#define CV_LOG_LEVEL_INFO 4
+#define CV_LOG_LEVEL_DEBUG 5
+#define CV_LOG_LEVEL_VERBOSE 6
+
+
+/**
+ * \def CV_LOG_STRIP_LEVEL
+ *
+ * Define CV_LOG_STRIP_LEVEL=CV_LOG_LEVEL_[DEBUG|INFO|WARN|ERROR|FATAL|DISABLED] to compile out anything at that and before that logging level
+ */
+#ifndef CV_LOG_STRIP_LEVEL
+# if defined NDEBUG
+#   define CV_LOG_STRIP_LEVEL CV_LOG_LEVEL_DEBUG
+# else
+#   define CV_LOG_STRIP_LEVEL CV_LOG_LEVEL_VERBOSE
+# endif
+#endif
+
+
+//! Supported logging levels and their semantic
+enum LogLevel {
+    LOG_LEVEL_SILENT = 0,              //!< for using in setLogVevel() call
```

Comment:
This header should not be included automatically with any of public OpenCV header file (like `opencv.hpp`), so serious problems are not expected.

BTW, CamelCase for "enumeration constants" violates [Naming conventions](https://github.com/opencv/opencv/wiki/Coding_Style_Guide#naming-conventions) guidelines.